RInAppMessaging/Classes/Views/ViewModel/FullViewModel.swift

```diff
@@ -11,4 +11,8 @@ internal struct FullViewModel {
     let isHTML: Bool
     let showOptOut: Bool
     let showButtons: Bool
+
+    var hasText: Bool {
+        [header, messageBody, messageBody].contains { $0?.isEmpty == false }
```

Comment:
messageBody twice?

Reply:
oh, nicely spotted. should be `lowerMessageBody`. I guess that qualifies for a unit test